### PR TITLE
dispatch button

### PR DESCRIPTION
**Pull request by raulgutierrez224**
adding dispatch button

### CODE REVIEW

**Review by micah-wehrle — COMMENTED** [review]
Just a few tweaks!

File: src/app/components/views/homepage/homepage.component.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { WeatherService } from 'src/app/services/weather.service';
 })
 export class HomepageComponent implements OnInit {
   public weatherAlertResponse: WeatherAlertResponse;
-  private ngUnsubscribe: Subject<void> = new Subject<void>;
+  private ngUnsubscribe: Subject<void> = new Subject<void>();
```
Comment (micah-wehrle) [inline src/app/components/views/homepage/homepage.component.ts:16]: This gave me an error! I had to add ()

File: src/app/services/job.service.ts [hunk 2]
```diff
@@ -26,6 +28,22 @@ export class JobService {
   public getJobs(): JobData[] {
     return this.jobs.slice();
   }
+
+  /**
+   * @description Will generate a new job and add it to the end of the current job array.
+   * @returns {void}
+   */
+  public generateNewJob(): void {
+    const newJob = new JobData(
```
Comment (looksliketyler) [inline src/app/services/job.service.ts:37]: data type

**Review by looksliketyler — APPROVED** [review]

File: src/app/services/job.service.ts [hunk 3]
```diff
@@ -26,6 +27,22 @@ export class JobService {
...
     return this.jobs.slice();
   }
+
+  /**
+   * @description Will generate a new job and add it to the end of the current job array.
+   * @returns {void}
+   */
+  public generateNewJob(): void {
+    const newJob: JobData = new JobData(
+      Math.round( (Math.random()*180 - 90) *1000) / 1000, // Generates a latitude between -90 and 90 with 3 decimal places
```
Comment (looksliketyler) [inline src/app/services/job.service.ts:37]: this approach needs changed as it can spit out invalid lat/long

**Comment by micah-wehrle** [comment]
branch merged a while ago, deleting branch

**Pull request merged**